nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/main/java/org/apache/nifi/authorization/ShellUserGroupProvider.java

```diff
@@ -194,10 +198,12 @@ public UserAndGroups getUserAndGroups(String identity) throws AuthorizationAcces
         logger.debug("Retrieved user {} for identity {}", new Object[]{user, identity});
 
         Set<Group> groups = new HashSet<>();
-
-        for (Group g : getGroups()) {
-            if (user != null && g.getUsers().contains(user.getIdentity())) {
-                groups.add(g);
+        if (user != null) {
+            for (Group g : getGroups()) {
+                if (user != null && g.getUsers().contains(user.getIdentifier())) {
```

Comment:
I'm not sure the null check needs to be repeated.

Reply:
Good point, I actually didn't notice the inner null check there.